### PR TITLE
[Example] Modify the output of gated_gcn example to fit benchmark test

### PR DESCRIPTION
**Pull request by caojy1998**
## Description

I modified the test output form for gated_gcn. I follow the modification pattern for gat, so that the test result can be read by python scripts in DGL_scripts.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/core/gated_gcn/train.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import time
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:1]: move it down, after the file description.

Comment (caojy1998) [reply examples/core/gated_gcn/train.py]: OK, done.

File: examples/core/gated_gcn/train.py [hunk 2]
```diff
@@ -98,13 +100,38 @@ def evaluate(model, device, data_loader, evaluator):
...
+        "--num_gpus",
+        type=int,
+        default=0,
+        help="Number of GPUs used for train and evaluation.",
+    )
+    args = parser.parse_args()
+    print(f"Training with DGL built-in GATConv module.")
 
     # Load ogb dataset & evaluator.
     dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:127]: why load ogb dataset instead of `args.dataset`?

File: examples/core/gated_gcn/train.py [hunk 3]
```diff
@@ -123,20 +150,21 @@ def main():
...
+        t1 = time.time()
         # Evaluate the prediction.
-        valid_acc = evaluate(model, device, valid_loader, evaluator)
-        test_acc = evaluate(model, device, test_loader, evaluator)
+        evaluate(model, device, valid_loader, evaluator)
+        acc = evaluate(model, device, test_loader, evaluator)
         print(
-            f"In epoch {epoch}, loss: {loss:.3f}, val acc: {valid_acc:.3f}, test"
-            f" acc: {test_acc:.3f}"
+            "Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} | Time {:.4f}".format(
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:165]: keep the val_acc printed as well.

File: examples/core/gated_gcn/train.py (hunk 2 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:127]: I will change it here. It seems that this model only supports some dataset start with `ogbn`.

File: examples/core/gated_gcn/train.py [hunk 4]
```diff
@@ -123,20 +150,21 @@ def main():
...
         # Kick off training.
+        t0 = time.time()
         loss = train(model, device, train_loader, opt, loss_fn)
-
+        t1 = time.time()
         # Evaluate the prediction.
-        valid_acc = evaluate(model, device, valid_loader, evaluator)
-        test_acc = evaluate(model, device, test_loader, evaluator)
+        evaluate(model, device, valid_loader, evaluator)
+        acc = evaluate(model, device, test_loader, evaluator)
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:163]: for test evaluate, we could just evaluate once, just after the train.

File: examples/core/gated_gcn/train.py [hunk 5]
```diff
@@ -123,20 +150,21 @@ def main():
...
-            f" acc: {test_acc:.3f}"
+            "Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} | Time {:.4f}".format(
+                epoch, loss, acc, t1 - t0
+            )
         )
 
-
-if __name__ == "__main__":
-    main()
+    print("Test accuracy {:.4f}".format(acc))
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:170]: add test evaluate here.

File: examples/core/gated_gcn/train.py (hunk 4 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:163]: OK, I will change the `Accuracy` during training as validate accuracy.

File: examples/core/gated_gcn/train.py (hunk 5 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:170]: OK, I have added it here.

File: examples/core/gated_gcn/train.py (hunk 3 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:165]: OK, I have changed the output here.

File: examples/core/gated_gcn/train.py [hunk 6]
```diff
@@ -98,12 +100,37 @@ def evaluate(model, device, data_loader, evaluator):
...
+        help="Number of epochs for train.",
+    )
+    parser.add_argument(
+        "--num_gpus",
+        type=int,
+        default=0,
+        help="Number of GPUs used for train and evaluation.",
+    )
+    args = parser.parse_args()
+    print(f"Training with DGL built-in GATConv module.")
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:124]: `f"` not required as nothing is formated

File: examples/core/gated_gcn/train.py [hunk 7]
```diff
@@ -98,12 +100,37 @@ def evaluate(model, device, data_loader, evaluator):
...
 
     # Load ogb dataset & evaluator.
-    dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
-    evaluator = Evaluator(name="ogbg-molhiv")
+    dataset = DglGraphPropPredDataset(name=args.dataset)
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:127]: are you sure datasets like `cora`, `pubmed` you specified in `args.dataset` supported by `DglGraphxxx` which is for graph property prediction for ogb datasets? Please make sure it really work.

Comment (Rhett-Ying) [reply examples/core/gated_gcn/train.py:127]: `GraphDataLoader` is instantiated.

**Review by Rhett-Ying — COMMENTED** [review]
if you change the default behavior of this script, update `README.md` accordingly.

File: examples/core/gated_gcn/train.py (hunk 7 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:127]: > are you sure datasets like `cora`, `pubmed` you specified in `args.dataset` supported by `DglGraphxxx` which is for graph property prediction for ogb datasets? Please make sure it really work.

Yes, I have tested locally. This API would download dataset and then load it.

File: examples/core/gated_gcn/train.py (hunk 6 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:124]: OK, I will delete it.

**Comment by caojy1998** [comment]
> if you change the default behavior of this script, update `README.md` accordingly.

OK, I have changed the `README.md`.

File: examples/core/gated_gcn/train.py (hunk 7 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py:127]: use ogbn dataset here.

File: examples/core/gated_gcn/train.py [hunk 8]
```diff
@@ -98,12 +100,37 @@ def evaluate(model, device, data_loader, evaluator):
...
 
-def main():
-    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default="cora",
+        help="Dataset name ('cora', 'citeseer', 'pubmed').",
```
Comment (Rhett-Ying) [inline examples/core/gated_gcn/train.py:109]: I don't think these datasets are supported in this example. please list the supported ones only.

**Review by Rhett-Ying — APPROVED** [review]

File: examples/core/gated_gcn/train.py (hunk 8 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py]: I have changed the default value and the help info here.

File: examples/core/gated_gcn/train.py [hunk 9]
```diff
@@ -125,18 +152,17 @@ def main():
...
-
+        t1 = time.time()
         # Evaluate the prediction.
-        valid_acc = evaluate(model, device, valid_loader, evaluator)
-        test_acc = evaluate(model, device, test_loader, evaluator)
+        val_acc = evaluate(model, device, valid_loader, evaluator)
         print(
-            f"In epoch {epoch}, loss: {loss:.3f}, val acc: {valid_acc:.3f}, test"
-            f" acc: {test_acc:.3f}"
+            "Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} | Time {:.4f}".format(
```
Comment (frozenbugs) [inline examples/core/gated_gcn/train.py:163]: Any reason not using f string?

File: examples/core/gated_gcn/train.py [hunk 10]
```diff
@@ -125,18 +152,17 @@ def main():
...
+                epoch, loss, val_acc, t1 - t0
+            )
         )
-
-
-if __name__ == "__main__":
-    main()
+    acc = evaluate(model, device, test_loader, evaluator)
+    print("Test accuracy {:.4f}".format(acc))
```
Comment (frozenbugs) [inline examples/core/gated_gcn/train.py:168]: ditto.

File: examples/core/gated_gcn/train.py (hunk 9 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py]: I will change it to f string here.

File: examples/core/gated_gcn/train.py (hunk 10 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py]: I will change it to f string here.

File: examples/core/gated_gcn/train.py [hunk 11]
```diff
@@ -125,18 +152,15 @@ def main():
...
-
+        t1 = time.time()
         # Evaluate the prediction.
-        valid_acc = evaluate(model, device, valid_loader, evaluator)
-        test_acc = evaluate(model, device, test_loader, evaluator)
+        val_acc = evaluate(model, device, valid_loader, evaluator)
         print(
-            f"In epoch {epoch}, loss: {loss:.3f}, val acc: {valid_acc:.3f}, test"
-            f" acc: {test_acc:.3f}"
+            f"Epoch {epoch:05d} | Loss {loss:.4f} | Accuracy {val_acc:.4f} | Time {t1 - t0:.4f}"
```
Comment (frozenbugs) [inline examples/core/gated_gcn/train.py:163]:
```suggestion
            f"Epoch {epoch:05d} | Loss {loss:.4f} | Accuracy {val_acc:.4f} | "
            f"Time {t1 - t0:.4f}"
```

**Review by frozenbugs — APPROVED** [review]

File: examples/core/gated_gcn/train.py (hunk 11 above)
Comment (caojy1998) [reply examples/core/gated_gcn/train.py]: OK, I have changed the format here.
